shanghai/network.py

```diff
@@ -68,38 +69,52 @@ def next_server(self):
     async def run(self):
         self.log_context.push()
 
-        def cancel_other_task_if_failed(task, other_task):
-            current_logger.info('cancel_other_task_if_failed', task)
-            if task.exception():
-                task.print_stack()
+        def cancel_other(task: asyncio.Task, other_task: asyncio.Task):
+            try:
```

Comment:
You're right. I realized only later that task.exception() does not raise an exception, if the task finished normally.